src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -0,0 +1,294 @@
+﻿// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Net;
+using Avro.Specific;
+using Avro.IO;
+using Confluent.SchemaRegistry;
+using System.Reflection;
+
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    ///     Avro specific serializer. Use this serializer with types generated
+    ///     using the avrogen.exe tool or one of the following primitive types: 
+    ///     int, long, float, double, boolean, string, byte[].
+    /// </summary>
+    /// <remarks>
+    ///     Serialization format:
+    ///       byte 0:           Magic byte use to identify the protocol format.
+    ///       bytes 1-4:        Unique global id of the avro schema that was used for encoding (as registered in Confluent Schema Registry), big endian.
+    ///       following bytes:  The serialized data.
+    /// </remarks>
+    public class AvroSerializer<T> : ISerializer<T>
+    {
+        private SpecificWriter<T> avroWriter;
+
+        private HashSet<string> topicsRegistered = new HashSet<string>();
+
+        private bool disposeClientOnDispose;
+
+        private string writerSchemaString { get; set; }
+
+        private const string InitialBufferSizePropertyName = "avro.buffer.bytes";
+
+        private const string AutoRegisterSchemaPropertyName = "avro.auto.register.schema";
```

Comment:
Where are these properties documented for the user?
And shouldnt it be schema.registry.auto.register.schemas?
Also, the Java client has a plural s at the end.

Reply:
they are documented right here :-). but good point, i didn't document because it wasn't clear exactly where to do so. we need better docs for this client, and they need to include this. Note that both properties are noted in the example which will go a long way, since most people will look at that.

good catch with the plural.

I think the properties shouldn't be schema.registry.[prop], since they relate to the schema registry client. 

I think better is `avro.serializer.auto.register.schemas`, since also they only relate to the serializer, no deserializer, and that makes that explicit, which is a good thing.